src/go/nerdctl-stub/command_handlers.go

```diff
@@ -22,15 +23,129 @@ func imageBuildHandler(c *commandDefinition, args []string) (*parsedArgs, error)
 		// input is a URL
 		return &parsedArgs{args: args}, nil
 	}
-	newPath, cleanups, err := filePathArgHandler(args[0])
+	newPath, cleanups, err := argHandlers.filePathArgHandler(args[0])
 	if err != nil {
-		for _, cleanup := range cleanups {
-			cleanupErr := cleanup()
-			if cleanupErr != nil {
-				log.Printf("Error cleaning up: %s", cleanupErr)
-			}
-		}
+		runCleanups(cleanups)
 		return nil, err
 	}
 	return &parsedArgs{args: append([]string{newPath}, args[1:]...), cleanup: cleanups}, nil
 }
+
+// hostPathResult is the return value of a hostPathDeterminerFunc that is used
+// in containerCopyHandler for determining which argument is the host path that
+// must be munged.
+type hostPathResult int
+
+const (
+	hostPathUnknown = hostPathResult(iota)
+	hostPathCurrent = hostPathResult(iota)
+	hostPathOther   = hostPathResult(iota)
+	hostPathNeither = hostPathResult(iota)
+)
+
+// containerCopyHandler handles `nerdctl container cp`
+func containerCopyHandler(c *commandDefinition, args []string, argHandlers argHandlersType) (*parsedArgs, error) {
+	var resultArgs []string
+	var cleanups []cleanupFunc
+	var paths []string
+
+	// Positional arguments `nerdctl container cp` are all paths, whether inside
+	// the container or outside.
+
+	for _, arg := range args {
+		if arg == "-" || !strings.HasPrefix(arg, "-") {
+			// If the arg is "-" (stdin/stdout) or doesn't start with -, it's a path.
+			paths = append(paths, arg)
+		} else {
+			resultArgs = append(resultArgs, arg)
+		}
+	}
+
+	if len(paths) != 2 {
+		// We should have exactly one source and one destination... just fail
+		runCleanups(cleanups)
+		return nil, fmt.Errorf("accepts 2 args, received %d", len(paths))
+	}
+
+	hostPathDeterminerFuncs := []func(i int, p string) (hostPathResult, error){
```

Comment:
None of the functions return an error; it is always `nil`, so the whole error handling can be removed.